lib/shopify_cli/partners_api/app_extensions/thread_pool.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module ShopifyCLI
+  class PartnersAPI
+    class AppExtensions
+      class ThreadPool
+        NUMBER_OF_THREADS = 10
+
+        def initialize
+          @jobs = Queue.new
+          @pool = Array.new(NUMBER_OF_THREADS) { spawn_thread }
+        end
+
+        def schedule(&block)
+          @jobs << block
```

Comment:
How are errors propagated when a block raises? In the case they are swallowed, I'd recommend storing them in the `thread_pool` and raising them when the caller calls `shutdown`. Also, what happens if one of the jobs fails? Are the other threads stopped?

Reply:
Great suggestion! 🚀 I've improved the mechanism a bit, and now jobs keep a state, so API users may raise errors after the thread pool consumes them.